File: docs/examples/simple.py
import sqlite3

# Create or connect to a SQLite database (a file named "mydb.db")
conn = sqlite3.connect("mydb.db")
cursor = conn.cursor()

# Create a table if it doesn't exist
cursor.execute('''
    CREATE TABLE IF NOT EXISTS users (
        id INTEGER PRIMARY KEY,
        name TEXT,
        age INTEGER
    )
''')

# Insert data into the table
cursor.execute("INSERT INTO users (name, age) VALUES (?, ?)", ("John Doe", 30))
cursor.execute("INSERT INTO users (name, age) VALUES (?, ?)",
               ("Jane Smith", 25))
conn.commit()  # Save the changes

# Retrieve data from the table
cursor.execute("SELECT * FROM users")
rows = cursor.fetchall()

# Display the retrieved data
for row in rows:
    print(f"ID: {row[0]}, Name: {row[1]}, Age: {row[2]}")

# Close the connection
conn.close()

File: docs/examples/simple.cpp
#include <QtSql>
#include <QCoreApplication>
#include <QDebug>

int main(int argc, char *argv[])
{
    QCoreApplication a(argc, argv);

    // Create an SQLite database
    QSqlDatabase db = QSqlDatabase::addDatabase("QSQLITE");
    db.setDatabaseName("mydb.db");

    if (!db.open()) {
        qDebug() << "Failed to open the database:" << db.lastError().text();
        return 1;
    }

    // Create a table
    QSqlQuery query;
    if (!query.exec("CREATE TABLE IF NOT EXISTS users (id INTEGER PRIMARY KEY, name TEXT, age INTEGER)")) {
        qDebug() << "Table creation failed:" << query.lastError().text();
        return 1;
    }

    // Insert data into the table
    query.prepare("INSERT INTO users (name, age) VALUES (?, ?)");
    query.addBindValue("John Doe");
    query.addBindValue(30);
    if (!query.exec()) {
        qDebug() << "Data insertion failed:" << query.lastError().text();
        return 1;
    }

    // Retrieve data from the table
    if (!query.exec("SELECT * FROM users")) {
        qDebug() << "Data retrieval failed:" << query.lastError().text();
        return 1;
    }

    while (query.next()) {
        int id = query.value(0).toInt();
        QString name = query.value(1).toString();
        int age = query.value(2).toInt();
        qDebug() << "ID:" << id << ", Name:" << name << ", Age:" << age;
    }

    // Close the database
    db.close();

    return 0;
}
